pkg/controller/daemon/daemon_controller.go

```diff
@@ -1535,3 +1571,7 @@ func isControlledByDaemonSet(p *v1.Pod, uuid types.UID) bool {
 	}
 	return false
 }
+
+func failedPodsBackoffKey(ds *apps.DaemonSet, nodeName string) string {
+	return fmt.Sprintf("%s/%d/%s", ds.UID, ds.Generation, nodeName)
```

Comment:
is `Generation` or `Status.ObservedGeneration` more correct here?

Reply:
I'd say `Generation` given that while processing the case where `Generation` just have been raised and using `ObservedGeneration` you would have just changed the key in the immediately following sync after raising `ObservedGeneration`.

Reply:
which seem like a good reason to use `ObservedGeneration` given we do rollingUpdate after, not before

(Also means I have to fix updating `ObservedGeneration` for DaemonSet because it shouldn't update it while waiting for expectations.)

I knew this was a tricky question :)

Reply:
for the record https://github.com/kubernetes/kubernetes/pull/67251

Reply:
> Also means I have to fix updating ObservedGeneration for DaemonSet because it shouldn't update it while waiting for expectations.

I'm not sure what that means... is the intent for the controller to not update observed generation until it has completed rollout of the new generation?

Reply:
What I meant there is that when DS is skipping control loop while waiting for expectations 
https://github.com/kubernetes/kubernetes/pull/67251/files#diff-3af9d435092399b4bf3ef61e0387eb36R1223
it shouldn't update `observedGeneration` as it didn't act on the particular `generation`.

unrelated; @liggitt do we agree on using `observedGeneration` here?

Reply:
are the pods a daemonset will create based on the observed generation or generation?

Reply:
> are the pods a daemonset will create based on the observedGeneration or generation?

generation

> but observedGeneration is still set to X-1

Not sure what you mean there. Controller acts on generation and when that's done it updates observedGeneration. That is meant for others to see that it has done all it should have to process that generation.

I bet there is a hidden meaning in your question, I just don't see it yet :)